/*
 * If not stated otherwise in this file or this component's LICENSE file the
 * following copyright and licenses apply:
 *
 * Copyright 2023 Sky UK
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

import "google/protobuf/descriptor.proto";
import "rialtoipc.proto";

package firebolt.rialto.ipc.transport;

message MethodCall {
  required uint64 serial_id = 1;
  required string service_name = 2;
  required string method_name = 3;

  optional bytes request_message = 4;
}

message MessageToServer {
  required MethodCall call = 1;
}

message MethodCallReply {
  required uint64 reply_id = 1;
  optional bytes reply_message = 2;
}

message MethodCallError {
  required uint64 reply_id = 1;
  required string error_reason = 2;
}

message EventFromServer {
  required string event_name = 1;
  optional bytes message = 2;
}

message MessageFromServer {
  oneof type {
    MethodCallReply reply = 1;
    MethodCallError error = 2;
    EventFromServer event = 3;
  }
}
